instrumentation/src/com/intellij/rt/coverage/instrumentation/Instrumenter.java

```diff
@@ -66,10 +66,15 @@ public MethodVisitor visitMethod(final int access,
     if (myEnum && isDefaultEnumMethod(name, desc, signature, myClassName)) {
       return mv;
     }
+    if (isDefaultConstructorOfSealedClass(access, name)) return mv;
     myProcess = true;
     return createMethodLineEnumerator(mv, name, desc, access, signature, exceptions);
   }
 
+  private boolean isDefaultConstructorOfSealedClass(int access, String name) {
+    return (access & Opcodes.ACC_SYNTHETIC) != 0 && name.equals("<init>");
```

Comment:
`@lombok.NoArgConstructor` would generate exactly this, is it good that it would be ignored as well? What are the executed lines in this `<init>`?

Reply:
While discussing this we decided that we should ignore all synthetic methods, but it seems like a good idea to test default constructor if a user creates it on purpose(even with `@lombok.NoArgConstructor`).
I can distinguish this particular case with more precise condition - constructor of sealed class has `kotlin/jvm/internal/DefaultConstructorMarker` parameter.